File: 782-jewels-and-stones/jewels-and-stones.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> st;
        for(char c : jewels) {
            st.insert(c);
        }
        int ans = 0;
        for(char c : stones) {
            if(st.count(c)) {
                ans++;
            }
        }
        return ans;
    }
};

File: 782-jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
    HashSet<Character> st = new HashSet<>();

    int count=0;
    char[] a1=jewels.toCharArray();
    char[] a2=stones.toCharArray();

    for(char x:a1)
    {
        st.add(x);
    }

    for(char x2:a2)
    {
        if(st.contains(x2))
        {
            count++;
        }
    }

    return count;


        
    }
}